pkg/destroy/openstack/openstack.go

```diff
@@ -1230,6 +1233,57 @@ func deleteVolumes(opts *clientconfig.ClientOpts, filter Filter, logger logrus.F
 	return true, nil
 }
 
+func deleteShares(opts *clientconfig.ClientOpts, filter Filter, logger logrus.FieldLogger) (bool, error) {
+	logger.Debug("Deleting OpenStack shares")
+	defer logger.Debugf("Exiting deleting OpenStack shares")
+
+	var clusterID string
+	for k, v := range filter {
+		if strings.ToLower(k) == "openshiftclusterid" {
+			clusterID = v
+			break
+		}
+	}
+
+	conn, err := clientconfig.NewServiceClient("sharev2", opts)
+	if err != nil {
+		logger.Error(err)
+		return false, nil
+	}
+
+	listOpts := shares.ListOpts{
+		Metadata: map[string]string{manilaCSIClusterIDKey: clusterID},
+	}
+
+	allPages, err := shares.ListDetail(conn, listOpts).AllPages()
+	if err != nil {
+		logger.Error(err)
+		return false, nil
+	}
+
+	allShares, err := shares.ExtractShares(allPages)
+	if err != nil {
+		logger.Error(err)
+		return false, nil
+	}
+
+	for _, share := range allShares {
+		logger.Debugf("Deleting share %q", share.ID)
+		err = shares.Delete(conn, share.ID).ExtractErr()
```

Comment:
If a share has one or more snapshots, this operation will fail; but since the cluster's being deleted anyway, should we delete snapshots first?

Reply:
you are right! I updated my patch to delete snapshots as well